pandas/core/generic.py

```diff
@@ -7680,8 +7693,42 @@ def between_time(
         if not isinstance(index, DatetimeIndex):
             raise TypeError("Index must be DatetimeIndex")
 
+        # If user has passed any of the depreciated arguments
+        # ('include_start', 'include_end') AND the new argument ('inclusive')
+        if (include_start != lib.no_default or include_end != lib.no_default) and (
+            inclusive is not None
+        ):
+            raise ValueError(
+                "Depreciated arguments `include_start` and `include_end`"
+                "cannot be passed if `inclusive` has been given."
+            )
+        # If any of the depreciated arguments ('include_start', 'include_end')
+        # have been passed
+        elif (include_start != lib.no_default) or (include_end != lib.no_default):
+            warnings.warn(
+                "`include_start` and `include_end` are deprecated in"
+                "favour of `inclusive`.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            if include_start == lib.no_default:
+                include_start = True
+            if include_end == lib.no_default:
+                include_end = True
+        else:  # On arg removal inclusive can default to "both"
+            if inclusive is None:
+                inclusive = "both"
+            elif inclusive not in ["both", "neither", "left", "right"]:
+                raise ValueError(
+                    "Inclusive has to be either string of 'both',"
+                    "'left', 'right', or 'neither'."
+                )
```

Comment:
can you also show what was received? So, if someone passes `inclusive='rihgt'` the error message would read

```
Inclusive has to be either string of 'both', 'left', 'right', or 'neither'. Got: 'rihgt'
```

Reply:
Yes, making the changes and updating the PR :)